### PR TITLE
fix: contains unknown publisher_handle columns.

### PR DESCRIPTION
**Pull request by hsgwa**
Signed-off-by: hsgwa <19860128+hsgwa@users.noreply.github.com>

closes https://github.com/tier4/caret/issues/54

### CODE REVIEW

File: src/caret_analyze/infra/lttng/records_provider_lttng.py [hunk 1]
```diff
@@ -486,6 +486,17 @@ def timer_records(self, timer: TimerStructValue) -> RecordsInterface:
 
         timer_events_factory = self._lttng.create_timer_events_factory(timer_lttng_cb)
         callback_records = self.callback_records(timer.callback)
+        if len(callback_records) == 0:
+            records = RecordsFactory.create_instance(
+                None,
+                columns=[
+                    ColumnValue(COLUMN_NAME.TIMER_EVENT_TIMESTAMP),
+                    ColumnValue(COLUMN_NAME.CALLBACK_START_TIMESTAMP),
+                    ColumnValue(COLUMN_NAME.CALLBACK_END_TIMESTAMP),
```
Comment (rokamu623) [inline src/caret_analyze/infra/lttng/records_provider_lttng.py:495]: Columns are not in alphabetical order.
If there is no special reason for this, it should be corrected.

File: src/test/infra/lttng/test_latency_definitions.py [hunk 2]
```diff
@@ -1256,6 +1298,53 @@ def test_inter_proc(
...
+
+        data = Ros2DataModel()
+        data.finalize()
+
+        publisher = create_publisher_struct()
+        subscription = create_subscription_struct()
+        callback = subscription.callback
+        communication = create_comm_struct(publisher, subscription)
+        publisher_lttng = create_publisher_lttng(pub_handle)
+        sub_cb_lttng = create_subscription_lttng(callback_obj, 5, callback_obj)
```
Comment (rokamu623) [inline src/test/infra/lttng/test_latency_definitions.py:1324]: magic number.

File: src/test/infra/lttng/test_latency_definitions.py [hunk 3]
```diff
@@ -1495,6 +1621,54 @@ def _create_timer_struct(
...
+        mocker,
+        create_lttng,
+        create_callback_lttng,
+        create_callback_struct,
+        bridge_mock,
+        bridge_setup_get_callback,
+    ):
+        callback_obj = 5
+        callback_obj_ = 7
+        # pid, tid = 15, 16
```
Comment (rokamu623) [inline src/test/infra/lttng/test_latency_definitions.py:1635]: This seems like an unnecessary description.

**Review by rokamu623 — CHANGES_REQUESTED** [review]
@hsgwa  
I have commented on some minor concerns.
Also, please merge the main branch again.

Sorry for the delay in reviewing.

File: src/test/infra/lttng/test_latency_definitions.py (hunk 3 above)
Comment (hsgwa) [reply src/test/infra/lttng/test_latency_definitions.py]: addressed in :[c670001](https://github.com/tier4/CARET_analyze/pull/210/commits/c6700010b36b4d8e0ec50486f0f6b972bd11a59a)

File: src/test/infra/lttng/test_latency_definitions.py (hunk 2 above)
Comment (hsgwa) [reply src/test/infra/lttng/test_latency_definitions.py]: addressed in :[c670001](https://github.com/tier4/CARET_analyze/pull/210/commits/c6700010b36b4d8e0ec50486f0f6b972bd11a59a)

File: src/caret_analyze/infra/lttng/records_provider_lttng.py (hunk 1 above)
Comment (hsgwa) [reply src/caret_analyze/infra/lttng/records_provider_lttng.py:495]: The order of these columns is not alphabetical.
For columns representing system time, they are in chronological order from left to right.

1. timer wake up
2. callback start
callback end

File: src/test/infra/lttng/test_latency_definitions.py [hunk 4]
```diff
@@ -1256,6 +1298,55 @@ def test_inter_proc(
...
+
+        data = Ros2DataModel()
+        data.finalize()
+
+        publisher = create_publisher_struct()
+        subscription = create_subscription_struct()
+        callback = subscription.callback
+        communication = create_comm_struct(publisher, subscription)
+        publisher_lttng = create_publisher_lttng(pub_handle)
+        sub_cb_lttng = create_subscription_lttng(callback_obj, subscription_handle, callback_obj_intra)
```
Comment (rokamu623) [inline src/test/infra/lttng/test_latency_definitions.py:1326]: This line exceeds 100 characters.

**Review by rokamu623 — CHANGES_REQUESTED** [review]
Please pass pytest.

Other than the above, there are no problems.

File: src/test/infra/lttng/test_latency_definitions.py (hunk 4 above)
Comment (hsgwa) [reply src/test/infra/lttng/test_latency_definitions.py]: addressed [65719fb](https://github.com/tier4/CARET_analyze/pull/210/commits/65719fb1c28ad9334f00127e17d4fa77468c4341)

**Review by rokamu623 — APPROVED** [review]
LGTM!

**Pull request merged**